### PR TITLE
Resolving Regression Bugs - File Bugs on `regression_report` and properly reset state on duplicate. 

### PR DESCRIPTION
**Pull request by nharper285**
## Summary of the Pull Request

_What is this about?_
We were not filing (or updating) bugs for regression reports. 

Separately, there was a bug in our notification logic that prevented bugs from having their state reset. 
## PR Checklist
* [ ] Applies to work item: #3262, #3213
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

## Validation Steps Performed

_How does someone test & validate?_

### CODE REVIEW

File: src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs [hunk 1]
```diff
@@ -20,13 +20,20 @@ public class Ado : NotificationsBase, IAdo {
...
     public async Async.Task<OneFuzzResultVoid> NotifyAdo(AdoTemplate config, Container container, IReport reportable, bool isLastRetryAttempt, Guid notificationId) {
         var filename = reportable.FileName();
-        if (reportable is RegressionReport) {
-            _logTracer.LogInformation("ado integration does not support regression report. container:{Container} filename:{Filename}", container, filename);
-            return OneFuzzResultVoid.Ok;
+        Report? report;
+        if (reportable is RegressionReport regressionReport) {
+            if (regressionReport.CrashTestResult.CrashReport is not null) {
+                report = regressionReport.CrashTestResult.CrashReport;
+                _logTracer.LogInformation("parsing regression report for ado integration. report:{report}", report);
```
Comment (stishkin) [inline src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs:27]: have single log line ? so everything is better grouped in app insights ?

File: src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs [hunk 2]
```diff
@@ -321,7 +328,9 @@ public sealed class AdoConnector {
                 });
             }
 
-            var systemState = JsonSerializer.Serialize(item.Fields["System.State"]);
+            // the below was causing on_duplicate not to work
+            // var systemState = JsonSerializer.Serialize(item.Fields["System.State"]);
```
Comment (stishkin) [inline src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs:332]: maybe this was supposed to be Json.Desrialize<string>(item.Fields["System.State"]) ? 

@tevoinea

**Review by stishkin — APPROVED** [review]
I'd wait for Teo's feedback though before merging

File: src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs (hunk 1 above)
Comment (nharper285) [reply src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs:27]: Yup I actually meant to delete this.

File: src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs [hunk 3]
```diff
@@ -20,13 +20,19 @@ public class Ado : NotificationsBase, IAdo {
...
-        if (reportable is RegressionReport) {
-            _logTracer.LogInformation("ado integration does not support regression report. container:{Container} filename:{Filename}", container, filename);
-            return OneFuzzResultVoid.Ok;
+        Report? report;
+        if (reportable is RegressionReport regressionReport) {
+            if (regressionReport.CrashTestResult.CrashReport is not null) {
+                report = regressionReport.CrashTestResult.CrashReport;
+                _logTracer.LogInformation("parsing regression report for ado integration. container:{Container} filename:{Filename}", container, filename);
+            } else {
+                _logTracer.LogInformation("ado integration does not support this regression report. container:{Container} filename:{Filename}", container, filename);
```
Comment (Porges) [inline src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs:29]: We probably want this at a higher log level?

**Review by tevoinea — APPROVED** [review]

File: src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs (hunk 2 above)
Comment (tevoinea) [reply src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs:331]: We were debugging it together earlier and there's no serialization required

File: src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs (hunk 3 above)
Comment (nharper285) [reply src/ApiService/ApiService/onefuzzlib/notifications/Ado.cs:29]: Switched to LogError

**Pull request merged**